### PR TITLE
workflow templates: deploy previews

### PR DESCRIPTION
**Pull request by sckott**
Based on my experimentation with Netlify deploy previews, I wanted to see how workflow templates work, using this workflow as a guinea pig. 

We can put any workflow templates in here that anyone in the org can use in a repo - I actually don't think we can restrict it to this org - but doesn't matter. 

The only thing needed for a workflow template are two files with the same slug but different file extensions:

- something.yml
- something.properties.json

There's still a few things to figure out:

- [x] figure out where to put the docs - in a readme right now, but I'm not sure you can put a readme in the workflow-templates dir
- [x] for the actions bot cleanup job, figure out how to make it just use the current repo rather than having to specify - eliminates one step

docs: 
- https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization
- https://docs.github.com/en/actions/using-workflows/reusing-workflows
- https://earthly.dev/blog/github-actions-reusable-workflows/

links:
- https://thoughtbot.com/blog/automate-processes-with-reusable-github-workflows

### CODE REVIEW

**Comment by sckott** [comment]
@seankross ready for review now, thoughts?

File: workflow-templates/deploy-preview.properties.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+    "name": "Quarto Deploy Preview Workflow",
+    "description": "Quarto Deploy preview starter workflow.",
+    "iconName": "octicon search",
```
Comment (sckott) [inline workflow-templates/deploy-preview.properties.json:4]: we could have any icon here, or even an svg https://docs.github.com/en/actions/using-workflows/creating-starter-workflows-for-your-organization perhaps of the dasl or wilds logo?

File: workflow-templates/deploy-preview.yml [hunk 2]
```diff
@@ -0,0 +1,67 @@
+on:
+  pull_request:
+    branches: [ $default-branch ]
```
Comment (sckott) [inline workflow-templates/deploy-preview.yml:3]: `$default-branch` picks up whatever default branch the repo has set

File: workflow-templates/deploy-preview.yml [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+        uses: quarto-dev/quarto-actions/setup@v2
+
+      - name: Render
+        uses: quarto-dev/quarto-actions/render@v2
+
+      - name: Deploy Preview to Netlify as preview
+        id: netlify-deploy
+        uses: nwtgck/actions-netlify@v2
+        env:
+          NETLIFY_SITE_ID: <your netlify id>
```
Comment (sckott) [inline workflow-templates/deploy-preview.yml:54]: This line is the only thing a user needs to modify in this file

File: workflow-templates/deploy-preview.properties.json (hunk 1 above)
Comment (seankross) [reply workflow-templates/deploy-preview.properties.json:4]: I'm confused about where this icon would show up but let's ship something so we can change it later.

File: workflow-templates/deploy-preview.yml (hunk 2 above)
Comment (seankross) [reply workflow-templates/deploy-preview.yml:3]: TIL, good to know

**Review by seankross — APPROVED** [review]
@sckott This is so cool! Green light to merge this immediately and then I want to test it out ASAP.

**Comment by sckott** [comment]
great, will do

File: workflow-templates/deploy-preview.properties.json (hunk 1 above)
Comment (sckott) [reply workflow-templates/deploy-preview.properties.json:4]: yep, can change later, i'll show you  if it's not clear

**Comment by sckott** [comment]
what it looks like now

Screenshot 2024-04-03 at 10 40 15 PM

**Pull request merged**